### ISSUE TITLE
ContentValueSetValidator

### ISSUE THREAD

**Issue by poornimanayar** [umbraco/Umbraco-CMS#15135]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.7

### Bug summary

It looks like the ability to add fields to exclude has been taken away from the constructors for [`ContentValueSetValidator`](https://apidocs.umbraco.com/v9/csharp/api/Umbraco.Cms.Infrastructure.Examine.ContentValueSetValidator.html)

Is this purposeful? 

### Steps to reproduce

See above

**Comment by bjarnef** [comment]
Hi @poornimanayar 

I don't think this has changed for a while.

This is how it looked based on an Umbraco v8.13.1 project. The `ContentValueSetValidator` only had overloads to include/exclude item types:

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/8db92566-5e72-4e05-a984-b8880859ec65)

On the other hand `MemberValueSetValidator` had an overload to exclude fields.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/2e012d00-99d8-43d9-ae48-f4455982ae3d)

**Comment by bjarnef** [comment]
This was based on Umbraco 8 though.

I seems to be different with latest codebase.
https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Infrastructure/Examine/ContentValueSetValidator.cs#L55

https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Infrastructure/Examine/ContentValueSetValidator.cs#L20-L62

It seems only one of the `Obsolete` methods pass in `includeFields` and `excludeFields` and the only non-obsolete method is mentioned used for tests:
https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Infrastructure/Examine/ContentValueSetValidator.cs#L21-L24

**Comment by bjarnef** [comment]
@Zeegaan it seems something is wrong `ContentValueSetValidator` - only the ctor for "tests" is non-obsolete:
https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Infrastructure/Examine/ContentValueSetValidator.cs#L21-L24

and no longer allow us to pass in `includeFields` and  `excludeFields`.

**Comment by poornimanayar** [comment]
@bjarnef Yes, I should have added those details into my original description but what you said is exactly the reason why I logged this issue. I can get rid of fields in `TransformingIndexValues`, but I think its should be ideally on the `ContentValueSetValidator`

**Comment by bjarnef** [comment]
Yes, I think it can't just obsolete this method:
https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Infrastructure/Examine/ContentValueSetValidator.cs#L44-L62

unless a new overload accept `Infrastructure.Scoping.ScopeProvider`.

It seems the constructors currently accept `Umbraco.Cms.Core.Scoping.ScopeProvider`.

**Comment by iOvergaard** [comment]
Hi there,
We're closing this issue as the affected version has since gone out of support — you can see which versions are currently supported in our [support policy](https://umbraco.com/products/knowledge-center/long-term-support-and-end-of-life/).
That doesn't necessarily mean the problem is gone, but a lot has changed in the codebase since this was reported, and we're no longer shipping fixes for this version. If you're still seeing this on a currently supported version, please open a new issue referencing this one with updated reproduction steps, and we'll happily take a look.
Thanks for taking the time to report it in the first place — we do appreciate it, even when we can't get every issue over the finish line.
